core/tests/client.rs

```diff
@@ -0,0 +1,61 @@
+use solana_client::rpc_client::RpcClient;
+use solana_core::validator::new_validator_for_tests;
+use solana_sdk::commitment_config::CommitmentConfig;
+use solana_sdk::pubkey::Pubkey;
+use solana_sdk::signature::KeypairUtil;
+use solana_sdk::system_transaction;
+use std::fs::remove_dir_all;
+use std::thread::sleep;
+use std::time::Duration;
+
+#[test]
+fn test_rpc_client() {
+    solana_logger::setup();
+
+    let (server, leader_data, alice, ledger_path) = new_validator_for_tests();
+    let bob_pubkey = Pubkey::new_rand();
+
+    let client = RpcClient::new_socket(leader_data.rpc);
+
+    assert_eq!(
+        client.get_version().unwrap(),
+        format!("{{\"solana-core\":\"{}\"}}", env!("CARGO_PKG_VERSION"))
+    );
+
+    assert_eq!(client.get_balance(&bob_pubkey).unwrap(), 0);
+
+    assert_eq!(client.get_balance(&alice.pubkey()).unwrap(), 10000);
+
+    let (blockhash, _fee_calculator) = client.get_recent_blockhash().unwrap();
+
+    let tx = system_transaction::transfer(&alice, &bob_pubkey, 20, blockhash);
+    let tx_response = client.send_transaction(&tx).unwrap();
+    let signature = tx.signatures[0].to_string();
+    let signature_str = signature.as_str();
+
+    assert_eq!(tx_response, signature);
+
+    let mut confirmed_tx = false;
+
+    for _ in 0..solana_sdk::clock::DEFAULT_TICKS_PER_SLOT {
```

Comment:
Using DEFAULT_TICKS_PER_SLOT in the loop confuses me as transaction confirmation time really has no direct correlation with ticks per slot.  How about changing the loop to be time based: 
```rust
let now = Instant::now();
while now.elapsed().as_secs() <= 10 {
    ..
    sleep(Duration::from_millis(500));
}